src/main/java/seedu/address/logic/commands/EditNoteCommand.java

```diff
@@ -0,0 +1,226 @@
+            + "[" + PREFIX_TIME + "TIME<HHMM>] "
+            + "[" + PREFIX_NOTE + "NOTE] "
+            + "Example: " + COMMAND_WORD + " 1 1 "
+            + PREFIX_DATE + "19-02-2024 "
+            + PREFIX_TIME + "1430 "
+            + PREFIX_NOTE + "General Flu ";
+
+    public static final String MESSAGE_EDIT_NOTE_SUCCESS = "Edited Note: %1$s";
+    public static final String MESSAGE_NOTE_NOT_EDITED = "No fields edited. Please try again.";
+    private final Index patientIndex;
+    private final Index noteIndex;
+    private final EditNoteDescriptor editNoteDescriptor;
+
+    /**
+     * @param patientIndex         of the patient with the appointment note to edit
+     * @param noteIndex            of the note in the filtered note list to edit
+     * @param editNoteDescriptor   details to edit the note with
+     */
+    public EditNoteCommand(Index patientIndex, Index noteIndex, EditNoteDescriptor editNoteDescriptor) {
+        requireNonNull(patientIndex);
+        requireNonNull(noteIndex);
+        requireNonNull(editNoteDescriptor);
+
+        this.patientIndex = patientIndex;
+        this.noteIndex = noteIndex;
+        this.editNoteDescriptor = editNoteDescriptor;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownPatientList = model.getFilteredPersonList();
+
+        if (patientIndex.getZeroBased() >= lastShownPatientList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_INDEX);
+        }
+
+        if (noteIndex.getZeroBased() >= lastShownPatientList.get(patientIndex.getZeroBased()).getNotes().size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_INDEX);
+        }
+
+        Person personToEdit = lastShownPatientList.get(patientIndex.getZeroBased());
+        Note noteToEdit = personToEdit.getNotes().get(noteIndex.getZeroBased());
+        Note editedNote = createEditedNote(noteToEdit, editNoteDescriptor);
+
+        ObservableList<Note> updatedNotes = FXCollections.observableArrayList(personToEdit.getNotes());
+        updatedNotes.set(noteIndex.getZeroBased(), editedNote);
+
+        Person updatedPerson = new Person.Builder(personToEdit).setNotes(updatedNotes).build();
+        model.setPerson(personToEdit, updatedPerson);
+
+        model.updateFilteredNoteList(PREDICATE_SHOW_ALL_NOTES);
+        return new CommandResult(String.format(MESSAGE_EDIT_NOTE_SUCCESS, Messages.format(editedNote)));
+    }
+
+    @Override
+    public String getCommandWord() {
+        return COMMAND_WORD;
+    }
+
+    @Override
+    public String getMessageUsage() {
+        return MESSAGE_USAGE;
+    }
+
+    /**
+     * Creates and returns a {@code Note} with the details of {@code noteToEdit}
+     * edited with {@code editNoteDescriptor}.
+     */
+    private static Note createEditedNote(Note noteToEdit, EditNoteDescriptor editNoteDescriptor) {
```

Comment:
Good job following the usage of edit command descriptor. Maybe if you have time you could also see how Person builder can be leveraged into notes via notes builder? But not really a must in this case, as descriptor would suffice.